### PR TITLE
SCALRCORE-21436 [provider] Scalr Provider > Provider configuration resource and data source 

### PR DESCRIPTION
**Pull request by DayS1eeper**
### Changelog
* [ ] I have updated CHANGELOG.md according to [CONTRIBUTING.md](../CONTRIBUTING.md)

### Documentation
* [ ] I have updated resource/data source documentation in [docs](../docs)

### State
* [ ] My changes affect the state
* [ ] I have included a state migration and a unit test for it

### CODE REVIEW

**Review by brezitskiy — APPROVED** [review]

**Review by artemvang — APPROVED** [review]

**Comment by penja** [comment]
@DayS1eeper please add docs that describes datasorce and resources. Ty

File: docs/data-sources/scalr_provider_configurations.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+---
+layout: "scalr"
+page_title: "Scalr: scalr_provider_configurations"
```
Comment (penja) [inline docs/data-sources/scalr_provider_configurations.md:3]: @DayS1eeper @emocharnik Please tell me more about the use case for this data source

File: scalr/resource_scalr_provider_configuration.go [hunk 2]
```diff
@@ -0,0 +1,487 @@
...
+
+		aws := v.([]interface{})[0].(map[string]interface{})
+		if access_key, ok := aws["access_key"].(string); ok {
+			configurationOptions.AwsAccessKey = scalr.String(access_key)
+		}
+		if secret_key, ok := aws["secret_key"].(string); ok {
+			configurationOptions.AwsSecretKey = scalr.String(secret_key)
+		}
+
+	} else if v, ok := d.GetOk("google"); ok {
```
Comment (penja) [inline scalr/resource_scalr_provider_configuration.go:183]: @DayS1eeper check please if we clear option if we change provider from gce to aws

**Review by penja — APPROVED** [review]

File: docs/data-sources/scalr_provider_configurations.md (hunk 1 above)
Comment (emocharnik) [reply docs/data-sources/scalr_provider_configurations.md:3]: The main idea behind having listing dat-source is to fetch all required at once (e.g for BU-1) and link it to respective environment

Comment (emocharnik) [reply docs/data-sources/scalr_provider_configurations.md:3]: Now to achieve that users do multiple requests to the api server

Comment (penja) [reply docs/data-sources/scalr_provider_configurations.md:3]: @emocharnik ok ty.

File: scalr/resource_scalr_provider_configuration.go (hunk 2 above)
Comment (DayS1eeper) [reply scalr/resource_scalr_provider_configuration.go:183]: @penja provider can't be changed, `provider_name` is not updatable in API, also error is raised on scale provider side if `provider_name` (e.g.  `gce` -> `aws`) is changed
https://github.com/Scalr/terraform-provider-scalr/pull/127/files#diff-e7c0b438989841ddb4feebb24352000b23756131f98d780821ff92619a91ce8eR29

**Pull request merged**